### DISCUSSION TITLE
Initial Focus in Dialog with Content inside Slot

### DISCUSSION THREAD

**Discussion by timgoeller** [tailwindlabs/headlessui#481] [Help]
I've extended the Dialog component with general styling, so I can reuse it in my project. Because of that, my `Modal` component, which includes the `Dialog` component uses a slot for its content. However, there are some cases where I want to set the `initialFocus`. 

There seems to be no straightforward / modular way to get the ref of the element I want to focus initially out of the slot. I think it would be very nice if `tabindex` was supported, so that the `Dialog`-content can handle tab-order freely. What would be the best-practice way to accomplish this at the moment?

**Comment by rrrrando** [comment]
Had the same problem today. Just build an `onMounted()` hook on the component you're sloting in to focus on the element you wish.
